### PR TITLE
Add QgsCodeEditorR code editor subclass for R scripts

### PR DESCRIPTION
**Pull request by nyalldawson**
Not used in QGIS itself anywhere, but added so that it can be used in plugins (specifically the R processing provider). 

Here's how it looks in the code editor preview pane:

image

### CODE REVIEW

**Comment by nyalldawson** [comment]
FYI @JanCaha @mdsumner

File: src/gui/codeeditors/qgscodeeditorr.h [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/***************************************************************************
+    qgscodeeditorjs.h - A R stats editor based on QScintilla
```
Comment (nirvn) [inline src/gui/codeeditors/qgscodeeditorr.h:2]:
```suggestion
    qgscodeeditorr.h - A R stats editor based on QScintilla
```

File: src/ui/qgscodeditorsettings.ui [hunk 2]
```diff
@@ -42,7 +42,7 @@
        <item>
         <widget class="QStackedWidget" name="mPreviewStackedWidget">
          <property name="currentIndex">
-          <number>4</number>
+          <number>0</number>
```
Comment (nirvn) [inline src/ui/qgscodeditorsettings.ui:45]: Was that deliberate, or something QtCreator accidentally changed? Is the new value correct (or are we just overriding this in the cpp? :))

**Review by nirvn — APPROVED** [review]

File: src/ui/qgscodeditorsettings.ui (hunk 2 above)
Comment (nyalldawson) [reply src/ui/qgscodeditorsettings.ui:45]: It's overriden, but seems nicer to default to first page in the ui file

**Pull request merged**